### PR TITLE
Add support for computing the reciprocal square root

### LINKED ISSUE #221: RFC: Add support for computing the reciprocal square root (rsqrt)
## Checklist

> Please ensure the following tasks are completed before submitting a feature request.

-   [x] Read and understood the [Code of Conduct][code-of-conduct].
-   [x] Searched for existing issues and pull requests.
-   [x] The issue name begins with `RFC:`.

## Description

> Description of the feature request.

This RFC proposes adding support for computing the reciprocal square root (i.e., 1/sqrt(x)). The implementation is trivial:

```javascript
/**
* @param {number} x - input value
* @returns {number} reciprocal square root
*/
function rsqrt( x ) {
    return 1.0 / sqrt( x );
}
```

The rationale behind providing this API is to provide a declarative interface, rather than imperatively writing out the intended behavior. As the implementation is small, modern compilers should readily inline the implementation, thus not affecting performance if used within other implementations.

Package: `@stdlib/math/base/special/rsqrt`
Alias: `rsqrt`.

## Related Issues

> Does this feature request have any related issues?

No.

## Questions

> Any questions for reviewers?

No.

## Other

> Any other information relevant to this feature request? This may include screenshots, references, sample output, and/or implementation notes.

No.

[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md

### PR DESCRIPTION
**Pull request by MatthewCochrane**
Resolves #221 .

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing], including the relevant style guides.
-   [x] Read and understand the [Code of Conduct][code-of-conduct].
-   [x] Read and understood the [licensing terms][license].
-   [x] Searched for existing issues and pull requests **before** submitting this pull request.
-   [x] Filed an issue (or an issue already existed) **prior to** submitting this pull request.
-   [x] Rebased onto latest `develop`.
-   [x] Submitted against `develop` branch.

## Description

> What is the purpose of this pull request?

This pull request:
- Add support for computing the reciprocal square root (i.e., 1/sqrt(x))

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #221 

## Questions

> Any questions for reviewers of this pull request?

I haven't added any benchmarks other than the js benchmark.  When I ran `make BENCHMARKS_FILTER=.*/math/base/special/sqrt/.* benchmark` (for `sqrt` which has other benchmarks than the javascript ones in `benchmark.js`) it would only run the javascipt benchmarks only.  
Is there any need for other language benchmarks at this point if they're not automatically run?

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

As I discussed on gitter briefly, I haven't made the markdown equations.  When I tried to run `markdown-equations` I got an error stating there was an unbound variable.

This is my first PR for this repo and the style guides and procedure are pretty involved.  Hopefully I've followed everything correctly but I'll be happy for you to point out any mistakes I've made.

Thanks a lot,
Matt

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md

[license]: https://github.com/stdlib-js/stdlib/blob/develop/LICENSE

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/rsqrt/README.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+</section>
+
+<!-- /.intro -->
+
+<section class="usage">
+
+## Usage
+
+```javascript
+var sqrt = require( '@stdlib/math/base/special/rsqrt' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rsqrt/README.md]:
```suggestion
var rsqrt = require( '@stdlib/math/base/special/rsqrt' );
```

File: lib/node_modules/@stdlib/math/base/special/rsqrt/README.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+<section class="examples">
+
+## Examples
+
+<!-- eslint no-undef: "error" -->
+
+```javascript
+var randu = require( '@stdlib/random/base/randu' );
+var round = require( '@stdlib/math/base/special/round' );
+var rsqrt = require( '@stdlib/math/base/special/sqrt' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rsqrt/README.md]:
```suggestion
var rsqrt = require( '@stdlib/math/base/special/rsqrt' );
```

File: lib/node_modules/@stdlib/math/base/special/rsqrt/lib/rsqrt.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+
+// MAIN //
+
+/**
+* Compute the reciprocal square root.
+*
+* @param {number} x - input value
+* @returns {number} reciprocal square root
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rsqrt/lib/rsqrt.js]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/rsqrt/lib/rsqrt.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+*
+* @example
+* v = rsqrt( -4.0 );
+* // returns NaN
+*
+* @example
+* v = rsqrt( NaN );
+* // returns NaN
+*/
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rsqrt/lib/rsqrt.js]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/rsqrt/lib/rsqrt.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+// MODULES //
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rsqrt/lib/rsqrt.js]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/rsqrt/lib/rsqrt.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rsqrt/lib/rsqrt.js]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/rsqrt/test/test.js [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+
+// MODULES //
+
+var tape = require( 'tape' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var PINF = require( '@stdlib/constants/math/float64-pinf' );
+var NINF = require( '@stdlib/constants/math/float64-ninf' );
+var EPS = require( '@stdlib/constants/math/float64-eps' );
+var abs = require( '@stdlib/math/base/special/abs' );
+var linspace = require( '@stdlib/math/utils/linspace' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rsqrt/test/test.js]:
```suggestion
```

TMK `linspace` is not used in this file.

File: lib/node_modules/@stdlib/math/base/special/rsqrt/README.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
+<!--
+
+@license Apache-2.0
+
+Copyright (c) 2018 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rsqrt/README.md]:
```suggestion
Copyright (c) 2019 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/rsqrt/benchmark/benchmark.js [hunk 9]
```diff
@@ -0,0 +1,72 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rsqrt/benchmark/benchmark.js]:
```suggestion
* Copyright (c) 2019 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/rsqrt/examples/index.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rsqrt/examples/index.js]:
```suggestion
* Copyright (c) 2019 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/rsqrt/lib/index.js [hunk 11]
```diff
@@ -0,0 +1,55 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rsqrt/lib/index.js]:
```suggestion
* Copyright (c) 2019 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/rsqrt/lib/rsqrt.js [hunk 12]
```diff
@@ -0,0 +1,69 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rsqrt/lib/rsqrt.js]:
```suggestion
* Copyright (c) 2019 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/rsqrt/test/fixtures/julia/runner.jl [hunk 13]
```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env julia
+#
+# @license Apache-2.0
+#
+# Copyright (c) 2018 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rsqrt/test/fixtures/julia/runner.jl]:
```suggestion
# Copyright (c) 2019 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/rsqrt/test/test.js [hunk 14]
```diff
@@ -0,0 +1,271 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rsqrt/test/test.js]:
```suggestion
* Copyright (c) 2019 The Stdlib Authors.
```

**Review by kgryte — CHANGES_REQUESTED** [review]
@MatthewCochrane This is awesome! Really well done. Some very minor nitpicks. Once resolved, this should be ready for merge! Thanks for working on this!

File: lib/node_modules/@stdlib/math/base/special/rsqrt/test/test.js (hunk 7 above)
Comment (MatthewCochrane) [reply lib/node_modules/@stdlib/math/base/special/rsqrt/test/test.js]: Yep, that's correct.  I added it for another test that I wrote and then ended up removing.

**Comment by MatthewCochrane** [comment]
Thanks @kgryte. Appreciate you taking the time to review it.  Good  catch on a few of those small mistakes with sqrt & rsqrt.

File: lib/node_modules/@stdlib/math/base/special/rsqrt/README.md [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+var v = rsqrt( 1.0 );
+// returns 1.0
+
+v = rsqrt( 4.0 );
+// returns 0.5
+
+v = rsqrt( 100.0 );
+// returns 0.1
+
+v = sqrt( 0.0 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rsqrt/README.md]:
```suggestion
v = rsqrt( 0.0 );
```

File: lib/node_modules/@stdlib/math/base/special/rsqrt/README.md [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+// returns Infinity
+
+v = sqrt( NaN );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rsqrt/README.md]:
```suggestion
v = rsqrt( NaN );
```

File: lib/node_modules/@stdlib/math/base/special/rsqrt/README.md [hunk 17]
```diff
@@ -0,0 +1,115 @@
...
+// returns NaN
+
+v = sqrt( Infinity );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rsqrt/README.md]:
```suggestion
v = rsqrt( Infinity );
```

**Review by kgryte — CHANGES_REQUESTED** [review]
@MatthewCochrane I missed a couple of lint errors in my initial review. Sorry about that. Once the example code is fixed, then I'll merge this PR! Thanks again for working on this!

**Review by kgryte — APPROVED** [review]

**Comment by kgryte** [comment]
@MatthewCochrane Merged! Thanks again for working on this!

**Comment by MatthewCochrane** [comment]
Not a problem @kgryte!

**Pull request merged**